src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/PetInfoOverlay.java

```diff
@@ -470,6 +470,10 @@ public static float getXpGain(Pet pet, float xp, String xpType) {
 				xp = xp * getBoostMultiplier(pet.petItem);
 			}
 		}
+		JsonObject pets = Constants.PETS;
+		if (pets != null && pets.has("custom_pet_leveling") && pets.get("custom_pet_leveling").getAsJsonObject().has(pet.petType.toUpperCase())) {
+			xp *= pets.get("custom_pet_leveling").getAsJsonObject().get(pet.petType.toUpperCase()).getAsJsonObject().get("xp_multiplier").getAsFloat();
```

Comment:
shouldnt you also check that xp_multiplier is present on that object? i feel like we could get a repo crash otherwise (+ having to specify the xp multiplier on every pet that has any special property feels like overkill)